airflow/operators/python.py

```diff
@@ -547,14 +547,21 @@ def __init__(
     ):
         if (
             python_version
-            and str(python_version)[0] != str(sys.version_info.major)
+            and python_version[0] != str(sys.version_info.major)
             and (op_args or op_kwargs)
         ):
             raise AirflowException(
                 "Passing op_args or op_kwargs is not supported across different Python "
                 "major versions for PythonVirtualenvOperator. Please use string_args."
                 f"Sys version: {sys.version_info}. Venv version: {python_version}"
             )
+        if type(python_version) is float or type(python_version) is int:
+            raise TypeError(
+                "Passing numeric type(int, float) as python_version "
+                "has been deprecated for PythonVirtualenvOperator. "
+                "Since Python 3.10 is interpreted as 3.1. "
+                "Please use string value instead."
+            )
```

Comment:
If we are raising error, it's not a deprecation

Reply:
We should handle this in such a way that it's not a breaking change

Reply:
Fixed it. thank you.